### PR TITLE
EmailNotification admin updates

### PR DESCRIPTION
**Pull request by lynnco**
- only add the SITE_ID when one is not already set (allows for easier editing of emails from the site you are not currently logged into)
- remove the current site when getting validation errors, instead of deleting it (which I thought wouldn't be an issue, but it turns out that its incredibly disruptive on admin)
- remove *all* invalid sites when invalid sites are found
- add test to ensure the email notification validation error includes enough information to resolve the error

### CODE REVIEW

File: callisto/notification/models.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def send(self, to, from_email, context=None):
         email.send()
 
     def add_site_from_site_id(self):
-        if getattr(settings, 'SITE_ID', None):
+        if not self.sites.count() and getattr(settings, 'SITE_ID', None):
```
Comment (kelseyq) [inline callisto/notification/models.py]: what does this do?

File: callisto/notification/validators.py [hunk 2]
```diff
@@ -3,9 +3,12 @@
...
         if EmailNotification.objects.filter(name=email.name, sites__id__in=[site.id]).exclude(id=email.id):
-            email.delete()
-            raise ValidationError('''
-                    EmailNotification already exists with (name={} ,site__domain={})
-                '''.format(email.name, site.domain))
+            invalid_sites.append(site.domain)
+            email.sites.remove(site.id)
+    if invalid_sites:
+        raise ValidationError('''
+                EmailNotification already exists with (name={}, sites__domain__in=[{}])
```
Comment (kelseyq) [inline callisto/notification/validators.py]: does this show up in admin UI?

File: callisto/notification/models.py (hunk 1 above)
Comment (lynnco) [inline callisto/notification/models.py]: only adds the SITE_ID to the current email if the current email doesn't have any sites

File: callisto/notification/validators.py (hunk 2 above)
Comment (lynnco) [inline callisto/notification/validators.py]: yep, shows up exactly like that

**Pull request merged**